### PR TITLE
Omit redundant `using: :btree` for schema dumping

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

**Review by rafaelfranca — APPROVED** [review]
This will need a CHANGELOG entry since it will change a public facing file. Could you add it?

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+*   Omit redundant `using: :btree` for schema dumping.
+
+    *Ryuta Kamizono*
```
Comment (kamipo) [inline activerecord/CHANGELOG.md]: I added a CHANGELOG entry!

**Comment by matthewd** [comment]
👍🏻 I prefer this as a ~solution for #26209. Ignoring values that might actually mean something worries me, whereas omitting the DBMS's native defaults (NB: not *configured* defaults) seems harmless -- and makes for a more streamlined schema file too.

... the generic schema dumper really shouldn't know about `:btree`, though :confused:

**Comment by rafaelfranca** [comment]
Yeah. When I reviewed I had the same option about the generic schema dumper knowing about btree but since postgresql and mysql have it I merged, but totally agree we should move to the index object if the using should be dumped or not. @kamipo willing to work on it?

**Comment by kamipo** [comment]
Yeah, I'm working on it here.

```diff
diff --git a/activerecord/lib/active_record/schema_dumper.rb b/activerecord/lib/active_record/schema_dumper.rb
index cf4495326b..63909ee3e7 100644
--- a/activerecord/lib/active_record/schema_dumper.rb
+++ b/activerecord/lib/active_record/schema_dumper.rb
@@ -184,12 +184,13 @@ def index_parts(index)
           index.columns.inspect,
           "name: #{index.name.inspect}",
         ]
+        default_index = @connection.default_index_type?(index)
         index_parts << "unique: true" if index.unique
         index_parts << "length: { #{format_options(index.lengths)} }" if index.lengths.present?
         index_parts << "order: { #{format_options(index.orders)} }" if index.orders.present?
         index_parts << "where: #{index.where.inspect}" if index.where
-        index_parts << "using: #{index.using.inspect}" if index.using && index.using != :btree
-        index_parts << "type: #{index.type.inspect}" if index.type
+        index_parts << "using: #{index.using.inspect}" if index.using && default_index
+        index_parts << "type: #{index.type.inspect}" if index.type && default_index
         index_parts << "comment: #{index.comment.inspect}" if index.comment
         index_parts
       end
```

**Comment by matthewd** [comment]
Might be simpler to have the adapter set e.g. `index.using` to `nil` when it's the default. (But I haven't really looked.)

**Comment by rafaelfranca** [comment]
Would not that hide information? I'd expect to using always return a value because the database always have that value.

**Comment by matthewd** [comment]
We're not providing a general database introspection service; these objects only exist for us to construct the schema dump.

**Pull request merged**